crates/node-api/src/payload_attributes.rs

```diff
@@ -0,0 +1,186 @@
+use crate::{validate_version_specific_fields, AttributesValidationError, EngineApiMessageVersion};
+use reth_primitives::{
+    revm::config::revm_spec_by_timestamp_after_merge,
+    revm_primitives::{BlobExcessGasAndPrice, BlockEnv, CfgEnv, SpecId},
+    Address, ChainSpec, Header, B256, U256,
+};
+use reth_rpc_types::engine::{
+    OptimismPayloadAttributes, PayloadAttributes as EthPayloadAttributes, PayloadId, Withdrawal,
+};
+
+/// This can be implemented by types that describe a currently running payload job.
+pub trait PayloadBuilderAttributes {
```

Comment:
I can see how it can be confusing that there are PayloadAttributes and Payload_Builder_Attributes.

so we should mention here that this intended as conversion type that transforms a PayloadAttr as received by the engine API to the type the builder can handle

Reply:
added a note